pretests/log_versions/log_versions.py

```diff
@@ -0,0 +1,50 @@
+r"""
+Summary
+-------
+
+Preserve docker versions in state files.
+
+Operational Summary
+-------------------
+
+#. Run 'docker version'; use DockerVersion module to parse output
+#. Run 'rpm -q docker|docker-latest' (whichever one is in use)
```

Comment:
Based on dealing with it in ADEPT, IMHO the server (daemon) versions is the most important.  You can determine which is being used by a zero exit status from ``systemctl --quiet is-active docker`` and ``systemctl --quiet is-active docker-latest``.  I'd suggest treating them equally, on the off-chance they're both enabled (a problem we need to know about).

In fact, maybe it would be good to just dump all the systemd unit details.  ``systemctl --plain --no-legend --no-pager list-unit-files`` will show everything along with it's enabled/disabled state.  ``systemctl --type=service --state=active --plain --no-legend --no-pager list-units | awk -e '{print $1" "$4}'``  will show the running/exited state of all active (started) services.

Hmmm, maybe this is too bug for just one subtest, maybe have separate sub-subtests for rpm vs systemd?